src/api/app/controllers/webui/users/subscriptions_controller.rb

```diff
@@ -1,5 +1,9 @@
 class Webui::Users::SubscriptionsController < Webui::WebuiController
-  before_action :require_login
+  # TODO: Remove this when we'll refactor kerberos_auth
+  before_action :kerberos_auth
+  before_action -> { authorize(EventSubscription.new) }
```

Comment:
Even though this controller is named `SubscriptionsController`, it's working with `EventSubscription::Form`, not `EventSubscription` (at least not directly).

Checking the `subscriber` of the `EventSubscription::Form` object seems like the right thing to do here. If `subscriber` is `nil` or the currently logged-in user, then they are authorized.

Reply:
Good suggestion, makes completely sense! I've introduced a `EventSubscription::FormPolicy`. In order to achieve that I had to introduce a few extra changes, LMK your your take on this 🙂